src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -327,6 +352,34 @@ protected Object invokeStep(StepDescriptor d, String name, Object args) {
         }
     }
 
+    private void logInterpolationWarnings(String stepName, @CheckForNull ArgumentsActionImpl argumentsAction, String nodeId, Set<String> interpolatedStrings, @CheckForNull EnvVars envVars, @Nonnull Set<String> sensitiveVariables, TaskListener listener) throws IOException, InterruptedException {
```

Comment:
I would go ahead and add a system property and check it in this method to control the behavior. I think `DSL.class.getName() + "UNSAFE_GROOVY_INTERPOLATION"` would be a decent name for the property, and `ignore`, `warn`, and `fail` could be good values representing the different behaviors, but feel free to come up with different names. Here is what I am thinking for the behavior of each option:

* `UNSAFE_GROOVY_INTERPOLATION=ignore`: Just return from this method like there were no secret variables.
* `UNSAFE_GROOVY_INTERPOLATION=warn`:  Default value. Matches the current behavior.
* `UNSAFE_GROOVY_INTERPOLATION=fail`: Instead of logging a warning, throws an `AbortException` so the Pipeline fails. The exception message is the same as what would be logged when using `warn`. Not as important as the other two options, we could omit this for now if desired.

Reply:
system property added in na24ffe0fb9cb442dede1dc5f54ad8e365d46dec4
Still need to add/update unit tests